### ISSUE TITLE
omni 2.0 version cache fails on portal

### ISSUE THREAD

**Issue by tcmitchell** [ahelsing/tickettest2#203]
I am using a pre-2.0 version of gcf for my development portal. When I try to run AM commands I get an error from omni because it cannot create a "~/.gcf" directory. The portal runs as user `www-data`, with home directory `/var/www`. `/var/www` is only writable by `root`, the directory creation fails.

Here is stack trace using sudo to simulate the environment:
```
$ sudo -u www-data -H /usr/share/geni-ch/portal/gcf/src/omni_php.py -c /tmp/tmitchel-omni.ini -l /usr/share/geni-ch/portal/gcf/src/logging.conf --logoutput /tmp/omni.log --api-version 2 -a https://localhost:8001/ --slicecredfile /tmp/geni-ULp9nt createsliver urn:publicid:IDN+dagoola:a1+slice+s2 /tmp/geni-41L5qX
Traceback (most recent call last):
  File "/usr/share/geni-ch/portal/gcf/src/omni_php.py", line 51, in <module>
    sys.exit(main())
  File "/usr/share/geni-ch/portal/gcf/src/omni_php.py", line 38, in main
    text, obj = omni.call( args, options )
  File "/home/tmitchel/gcf-pre-2.0/src/omni.py", line 423, in call
    return API_call( framework, config, args, opts, verbose=verbose )
  File "/home/tmitchel/gcf-pre-2.0/src/omni.py", line 440, in API_call
    result = handler._handle(args)
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/handler.py", line 80, in _handle
    return getattr(self.amhandler, call)(args[1:])
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 1306, in createsliver
    args)
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 420, in _api_call
    (ver, newc) = self._checkValidClient(client)
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 2925, in _checkValidClient
    cver, message = self._get_this_api_version(client)
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 351, in _get_this_api_version
    (res, message) = self._get_getversion_key(client, 'geni_api')
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 336, in _get_getversion_key
    (versionSpot, message) = self._get_getversion_value(client)
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 322, in _get_getversion_value
    (thisVersion, message) = self._do_and_check_getversion(client)
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 277, in _do_and_check_getversion
    (thisVersion, message) = self._do_getversion(client)
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 114, in _do_getversion
    self._cache_getversion(client, thisVersion, message)
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 224, in _cache_getversion
    self._save_getversion_cache()
  File "/home/tmitchel/gcf-pre-2.0/src/omnilib/amhandler.py", line 159, in _save_getversion_cache
    os.makedirs(fdir)
  File "/usr/lib/python2.6/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/var/www/.gcf'
```

 _Imported from trac ticket [#203](http://trac.gpolab.bbn.com/proto-ch/ticket/203),  created by tmitchel on 09-06-2012 at 14:30, last modified: 11-02-2012 at 16:39_

**Comment by tcmitchell** [comment]
We'll need to take care of this when we switch the portal to omni 2.0. Here are two alternatives, and there may be others:
 * make omni more flexible about creating the cache dir (possibly disabling the cache if the directory doesn't exist and can't be created)
 * create `/var/www/.gcf` at install time with appropriate permissions


_Trac comment by tmitchel (github user: tcmitchell) on 09-06-2012 at 14:38_

**Comment by ahelsing** [comment]
Note that the path to the cache dir is an option.
But your other options are good too.


_Trac comment by ahelsing on 09-06-2012 at 15:19_

**Comment by ahelsing** [comment]
We use 2.0 now. The cache is in /tmp.
Verify this if fully fixed.


_Trac comment by ahelsing on 10-30-2012 at 15:51_

**Comment by tcmitchell** [comment]
Fixed in late September, [ce2af388e8ca00cac65b078f434f107dd1c220eb](https://github.com/ahelsing/tickettest2/commit/ce2af388e8ca00cac65b078f434f107dd1c220eb)

_Trac comment by tmitchel (github user: tcmitchell) on 11-02-2012 at 16:39_
